src/Razor/src/Microsoft.AspNetCore.Razor.Language/NotFoundProjectItem.cs

```diff
@@ -16,10 +16,12 @@ internal class NotFoundProjectItem : RazorProjectItem
         /// </summary>
         /// <param name="basePath">The base path.</param>
         /// <param name="path">The path.</param>
-        public NotFoundProjectItem(string basePath, string path)
+        /// <param name="fileKind">The file kind</param>
+        public NotFoundProjectItem(string basePath, string path, string fileKind)
         {
             BasePath = basePath;
             FilePath = path;
+            FileKind = fileKind ?? FileKinds.GetComponentFileKindFromFilePath(path);
```

Comment:
Why `GetComponentFileKindFromFilePath` and not `GetFileKindFromFilePath`?

Reply:
Fat finger, good catch